### PR TITLE
errno is set to EINVAL on successful writes

### PR DESCRIPTION
**Pull request by prashanthpai**
Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

**Comment by prashanthpai** [comment]
Snippet to reproduce the bug: https://gist.github.com/prashanthpai/c89f096698faf89490cb

File: gfapi/fd.go [hunk 1]
```diff
@@ -109,6 +109,11 @@ func (fd *Fd) Read(b []byte) (int, error) {
 func (fd *Fd) Write(b []byte) (int, error) {
 
 	n, err := C.glfs_write(fd.fd, unsafe.Pointer(&b[0]), C.size_t(len(b)), 0)
+	if n == C.ssize_t(len(b)) {
+		// FIXME: errno is set to EINVAL even though write is successful. This
+		// is probably a bug. Remove this workaround when that gets fixed.
+		err = nil
+	}
```
Comment (kshlm) [inline gfapi/fd.go]: Is this a bug with gfapi itself? If there is a bug-id associated with this for gfapi, please mention it.

Comment (prashanthpai) [inline gfapi/fd.go]: Haven't looked into it yet but made a note of it and plan to do it later. I'm certain that the bug is not in the go bindings. The behavior is easily observable: https://gist.github.com/prashanthpai/c89f096698faf89490cb

**Comment by prashanthpai** [comment]
~~~
ret, err := C.glfs_set_logging(v.fs, nil, C.int(logLevel))
return int(ret), err
~~~

This also results in ret being 0, errno set to `ENOENT` and hence err returned is not `nil`.
Just making a note here so that I don't forget.

**Comment by kshlm** [comment]
I opened an issue #14 to track the workaround. I'll merge this now.

Thanks.

**Pull request merged**